src/Microsoft.Health.Dicom.Api/Features/Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -84,6 +84,7 @@ private IActionResult MapExceptionToResult(Exception exception)
             case AuditHeaderTooLargeException:
             case ConnectionResetException:
             case OperationCanceledException:
+            case DataStoreException e when e.InnerException is TaskCanceledException:
```

Comment:
Should we way `OperationCanceledException` instead?

Reply:
I think they come from different sources - the blob libraries throw `TaskCanceledException`.